### ISSUE TITLE
object-curly-newline multiline

### ISSUE THREAD

**Issue by hrazmsft** [eslint/eslint#14694]
**Tell us about your environment**

Node version: v15.12.0
npm version: v7.15.1
Local ESLint version: v7.22.0 (Currently used)
Global ESLint version: Not found

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?** default (came from CRA)

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
        "object-curly-newline": [
            "warn",
            {
                "ObjectExpression": { "multiline": true, "minProperties": 1 },
                "ObjectPattern": { "multiline": true, "minProperties": 5},
                "ImportDeclaration": { "multiline": true, "minProperties": 7},
                "ExportDeclaration": { "multiline": true, "minProperties": 7}
            }
        ]
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const item = { name: text, key: text };
```

```bash
eslint --ext .tsx,.ts,.jsx,.js --fix src/
```

**What did you expect to happen?**

Since we are using `multiline: true` with `--fix` flag, I would expect it to fix the code in a correct form. Like:

```
const item = { 
    name: text,
    key: text 
};
```

**What actually happened? Please copy-paste the actual, raw output from ESLint.**

The fixed code was:
```
const item = { 
    name: text, key: text 
};
```
Which is not as documented here https://eslint.org/docs/rules/object-curly-newline
Screenshot:
![image](https://user-images.githubusercontent.com/66222191/121526467-12421c80-ca02-11eb-9996-fa748c07218f.png)

**Steps to reproduce this issue:**

1. Add the configuration above to your ESLint configuration
1. Create an single line object with multiple entries
1. Run `eslint` with `fix` flag

**Are you willing to submit a pull request to fix this bug?** No

**Comment by mdjermanovic** [comment]
Hi @hrazmsft, thanks for the issue! 

> The fixed code was:
> 
> ```
> const item = { 
>     name: text, key: text 
> };
> ```
> 
> Which is not as documented here https://eslint.org/docs/rules/object-curly-newline

This is indeed incorrect code for `{ "multiline": true }` ([Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgb2JqZWN0LWN1cmx5LW5ld2xpbmU6IFtcImVycm9yXCIsIHsgXCJtdWx0aWxpbmVcIjogdHJ1ZSB9XSovXG5cbmNvbnN0IGl0ZW0gPSB7IFxuICAgIG5hbWU6IHRleHQsIGtleTogdGV4dCBcbn07Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMiwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==)), but you also have `"minProperties": 1` in the configuration for `"ObjectExpression"`.

If any of the conditions is `true` (multiline properties OR number of properties >= 1) , the rule will require linebreaks after `{` and before `}`.

> Since we are using `multiline: true` with `--fix` flag, I would expect it to fix the code in a correct form. Like:
> 
> ```
> const item = { 
>     name: text,
>     key: text 
> };
> ```

`object-curly-newline` requires/disallows only linebreaks right after `{` and right before `}`:

> This rule requires or disallows a line break between `{` and its following token, and between `}` and its preceding token of object literals or destructuring assignments.

You can also enable the [object-property-newline](https://eslint.org/docs/rules/object-property-newline) rule.

Here's a demo with multiple rules where the original example gets autofixed to the expected format (please take a look at the Fixed Code tab) - [Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgXCJvYmplY3QtY3VybHktbmV3bGluZVwiOiBbXG4gICAgXCJ3YXJuXCIsXG4gICAge1xuICAgICAgICBcIk9iamVjdEV4cHJlc3Npb25cIjogeyBcIm11bHRpbGluZVwiOiB0cnVlLCBcIm1pblByb3BlcnRpZXNcIjogMSB9LFxuICAgICAgICBcIk9iamVjdFBhdHRlcm5cIjogeyBcIm11bHRpbGluZVwiOiB0cnVlLCBcIm1pblByb3BlcnRpZXNcIjogNX0sXG4gICAgICAgIFwiSW1wb3J0RGVjbGFyYXRpb25cIjogeyBcIm11bHRpbGluZVwiOiB0cnVlLCBcIm1pblByb3BlcnRpZXNcIjogN30sXG4gICAgICAgIFwiRXhwb3J0RGVjbGFyYXRpb25cIjogeyBcIm11bHRpbGluZVwiOiB0cnVlLCBcIm1pblByb3BlcnRpZXNcIjogN31cbiAgICB9XG5dKi9cbi8qZXNsaW50IFwib2JqZWN0LXByb3BlcnR5LW5ld2xpbmVcIjogW1wid2FyblwiXSovXG4vKmVzbGludCBcImluZGVudFwiOiBbXCJ3YXJuXCIsIDRdICovXG5cbmNvbnN0IGl0ZW0gPSB7IG5hbWU6IHRleHQsIGtleTogdGV4dCB9OyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTIsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=)

**Comment by hrazmsft** [comment]
@mdjermanovic thanks for the comment. Actually, this example is one of many. It is happening with `import`, `export` and destructuring objects that `object-curly-newline` doesn't cover.

Any idea how to solve that in these cases?

I set the `minProperties` to 1 since I don't want a new line in empty objects...

**Comment by mdjermanovic** [comment]
> It is happening with import, export and destructuring objects that object-curly-newline doesn't cover.

`object-curly-newline` covers `import`, `export` and destructuring patterns as well, but `object-property-newline` indeed covers only object expressions.

Unfortunately, [stylistic rules are frozen](https://eslint.org/blog/2020/05/changes-to-rules-policies), so the best course of action would be to make a custom rule based on `object-property-newline` that would also cover `import`, `export` and destructuring.

**Comment by hrazmsft** [comment]
Thank you
